### PR TITLE
move username and groupname validation to apiserver-library-go

### PR DESCRIPTION
**Pull request by stlaz**
/assign @sttts 

/hold
contains fake bump

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -25,7 +26,6 @@ require (
 	k8s.io/client-go v0.19.2
 	k8s.io/code-generator v0.19.2
 	k8s.io/component-base v0.19.2
-	k8s.io/klog/v2 v2.3.0
```
Comment (sttts) [inline go.mod:28]: isn't go mod cool?

Comment (stlaz) [reply go.mod:28]: yeah, I noticed this change, too. It's... eh.

**Pull request merged**